pennylane/templates/layers/simplified_two_design.py

```diff
@@ -40,51 +35,24 @@ def _preprocess(weights, initial_layer_weights, wires):
     Returns:
         int: number of times that the ansatz is repeated
     """
+    shape = qml.math.shape(weights)
+    repeat = shape[0]
 
-    if qml.tape_mode_active():
-
-        shape = qml.math.shape(weights)
-        repeat = shape[0]
-
-        if len(shape) > 1:
-            if shape[1] != len(wires) - 1:
-                raise ValueError(
-                    f"Weights tensor must have second dimension of length {len(wires) - 1}; got {shape[1]}"
-                )
-
-            if shape[2] != 2:
-                raise ValueError(
-                    f"Weights tensor must have third dimension of length 2; got {shape[2]}"
-                )
+    if len(shape) > 1:
+        if shape[1] != len(wires) - 1:
+            raise ValueError(
+                f"Weights tensor must have second dimension of length {len(wires) - 1}; got {shape[1]}"
+            )
 
-        shape2 = qml.math.shape(initial_layer_weights)
-        if shape2 != (len(wires),):
+        if shape[2] != 2:
             raise ValueError(
-                f"Initial layer weights must be of shape {(len(wires),)}; got {shape2}"
+                f"Weights tensor must have third dimension of length 2; got {shape[2]}"
             )
 
-    else:
-        repeat = check_number_of_layers([weights])
-
-        expected_shape_initial = (len(wires),)
-        check_shape(
-            initial_layer_weights,
-            expected_shape_initial,
-            msg="Initial layer weights must be of shape {}; got {}"
-            "".format(expected_shape_initial, get_shape(initial_layer_weights)),
-        )
-
-        if len(wires) in [0, 1]:
-            expected_shape_weights = (0,)
-        else:
-            expected_shape_weights = (repeat, len(wires) - 1, 2)
-
-        check_shape(
-            weights,
-            expected_shape_weights,
-            msg="Weights tensor must be of shape {}; got {}"
-            "".format(expected_shape_weights, get_shape(weights)),
-        )
+    shape2 = qml.math.shape(initial_layer_weights)
+    if shape2 != (len(wires),):
+        raise ValueError(f"Initial layer weights must be of shape {(len(wires),)}; got {shape2}")
+
```

Comment:
On line 64-65 `qml.Variable` is referred to, in the `entangler` template docstring.

Reply:
💯